adapters/appnexus/appnexus.go

```diff
@@ -204,19 +207,171 @@ func (a *AppNexusAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder
 			}
 
 			mediaType := getMediaTypeForImp(bid.ImpID, anReq.Imp)
-			pbid.CreativeMediaType = mediaType
+			pbid.CreativeMediaType = string(mediaType)
 			bids = append(bids, &pbid)
 		}
 	}
 
 	return bids, nil
 }
-func getMediaTypeForImp(impId string, imps []openrtb.Imp) string {
-	mediaType := "banner"
+
+func (a *AppNexusAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	memberIds := make([]string, 0, len(request.Imp))
+	errs := make([]error, 0, len(request.Imp))
+
+	for i := 0; i < len(request.Imp); i++ {
+		memberId, err := preprocess(&request.Imp[i])
+		if memberId != "" {
+			memberIds = append(memberIds, memberId)
```

Comment:
Could consider having memberIds as a map[string]int, and change this line to be memberIds[memberId] = 1. Then later down you can len(memberIds) as a quick check if we have multiple Ids. You can move the scope of memberId outside the loop, so you can use the last memberId as the default member ID to use for the request.